src/ir/ty.rs

```diff
@@ -869,6 +874,60 @@ impl Type {
                                            parent_id,
                                            ctx);
             }
+            CXType_ObjCInterface => {
+
+
+                let itemid = Item::from_ty(ty, location, parent_id, ctx);
+                TypeKind::ObjCInterface(itemid.unwrap())
+                //                panic!("woohoo! itemid:{:?} ", itemid); // TODO: probably needs its own TypeKind
+            }
+            CXType_ObjCObjectPointer => {
+                //                panic!("yeah!");
+                println!("ty={:?}", ty);
+                println!("pointee_type={:?}", ty.pointee_type());
+
+                /*                let referenced = location.unwrap().referenced().unwrap();
+                let referenced_ty = referenced.cur_type();
+                let declaration = referenced_ty.declaration();
+                println!("referenced={:?}", referenced);
+                println!("referenced_ty={:?}", referenced_ty);
+                println!("declaration={:?}", declaration);
+                // println!("named={:?}", ty.named());
+                // find item already built item with the ObjCInterface?
+                println!("ty spelling={:?}", ty.spelling());
+                let param =
+                    Item::named_type(ty.pointee_type().unwrap().spelling(),
+                                     potential_id,
+                                     ctx);
+                println!("param named_type={:?}", param);
+                println!("element type ={:?}", ty.elem_type());
+                println!("typedef type ={:?}",
+                         location.unwrap().typedef_type());
+                println!("cur type ={:?}", location.unwrap().cur_type());
+                println!("objc encoding ={:?}", ty.objc_encoding());*/
+
+                let inner = Item::from_ty_or_ref(ty.pointee_type().unwrap(),
```

Comment:
This pretty much shares the body with `CXType_Pointer`, reference, et. al, right? If so, let's just fold it there when you're done with debugging it :)